### PR TITLE
Information preserving rotation

### PR DESCRIPTION
**Pull request by Jordan-Dennis**
Hi guys, 
This just takes the information preserving rotation that is discussed in #104 and adds it in `dlux/layers.py`. @LouisDesdoigts, I just want your opinion on the parameterisation.

Regards

Jordan

### CODE REVIEW

File: dLux/layers.py [hunk 1]
```diff
@@ -1030,3 +1031,163 @@ def CLIMB(self, wf, ppsz = 256):
...
+        padding: int
+            This parameter cannot be differentiated and will default to 4.
+            Change the parameter to 2 for a performance boost but it may 
+            not result in lower accuracy.
+        """
+        self.alpha = np.asarray(alpha).astype(float)
+        self.padding = int(padding)
+
+
+    def _rotate(image: float, alpha: float, pad: int = 4) -> float:
```
Comment (LouisDesdoigts) [inline dLux/layers.py:1073]: Change default padding to 2

File: dLux/layers.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 import jax.numpy as np
 import equinox as eqx
 import dLux
+from dLux.utils import polar2cart, cart2polar, get_pixel_positions
```
Comment (LouisDesdoigts) [inline dLux/layers.py:34]: Reference these in the code as `dLux.polar2cart` ... etc

**Comment by LouisDesdoigts** [comment]
Yeah this all looks good, just those two little changes and then passing the tests and I'll merge this in.
